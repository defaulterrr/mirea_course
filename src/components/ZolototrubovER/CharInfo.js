import React, {useState, useEffect} from 'react'
import Characters from './Character/Characters'

const fetchChar = async (stateUpdate) => {

    fetch("https://secure-hamlet-19722.herokuapp.com/api/v1/characters")
    .then(response => response.json())
    .then(json => {
        stateUpdate({
            characters: json,
            isLoading: false
        })
        console.log(json)
    }
    )
}

const CharInfo = (props) => {
    const [ state,setState ] = useState({
        characters: props.characters || null,
        isLoading: props.isLoading || true,
    })

    useEffect(() => {
        fetchChar(setState)
    },[])



    return (
        <div className="Characters">
            <p>Marvel vs. Capcom 2 character info</p>
            <table className='CharTable'>
                <th>Character</th>
                <th>Icon</th>
                <th>Universe</th>
                <th>Info</th>
                {state.isLoading === false ? state.characters.map((chunk)=> {
                    if(chunk.name.includes('Dalvin')) {}
                    else {
                    if(chunk.name.includes('Wolverine')) {
                      const info = "https://marvelvscapcom.fandom.com/wiki/Wolverine"
                      return (
                          <Characters name={chunk.name} head_shot={chunk.head_shot} universe={chunk.universe} loadLink={'https://secure-hamlet-19722.herokuapp.com/api/v1/characters/' + chunk.name}
                          info={info}/>
                         )}
                    else if(chunk.name.includes(' ')) {
                      const info = "https://marvelvscapcom.fandom.com/wiki/" + chunk.name.split(' ')[0] + '_' + chunk.name.split(' ')[1]
                      return (
                          <Characters name={chunk.name} head_shot={chunk.head_shot} universe={chunk.universe} loadLink={'https://secure-hamlet-19722.herokuapp.com/api/v1/characters/' + chunk.name}
                          info={info}/>
                        )} else {
                          const info = "https://marvelvscapcom.fandom.com/wiki/" + chunk.name
                          return (
                              <Characters name={chunk.name} head_shot={chunk.head_shot} universe={chunk.universe} loadLink={'https://secure-hamlet-19722.herokuapp.com/api/v1/characters/' + chunk.name}
                              info={info}/>
                        )}
                    }
                }) : null}
            </table>
        </div>
    )
}

export default CharInfo
