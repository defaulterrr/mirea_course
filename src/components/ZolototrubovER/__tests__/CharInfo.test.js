import React from 'react'
import { create } from 'react-test-renderer'
import CharInfo from '../CharInfo'

const testData = [{"name":"Blackheart","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/3/34/MVC2_Blackheart.jpg/revision/latest?cb=20110914071301","universe":"Marvel"},{"name":"Cable","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/b/be/MVC2_Cable.jpg/revision/latest?cb=20110914072054","universe":"Marvel"},{"name":"Captain America","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/2/21/MVC2_Captain_America.jpg/revision/latest?cb=20110914072107","universe":"Marvel"},{"name":"Colossus","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/3/31/MVC2_Colossus.jpg/revision/latest?cb=20110914072123","universe":"Marvel"},{"name":"Cyclops","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/c/c3/MVC2_Cyclops.jpg/revision/latest?cb=20110914072139","universe":"Marvel"},{"name":"Doctor Doom","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/b/ba/MVC2_Doctor_Doom.jpg/revision/latest?cb=20110914072159","universe":"Marvel"},{"name":"Gambit","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/f/f4/MVC2_Gambit.jpg/revision/latest?cb=20110914072215","universe":"Marvel"},{"name":"Hulk","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/1/19/MVC2_Hulk.jpg/revision/latest?cb=20110914072229","universe":"Marvel"},{"name":"Ice Man","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/c/cd/MVC2_Iceman.jpg/revision/latest?cb=20110914072244","universe":"Marvel"},{"name":"Iron Man","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/6/6b/MVC2_Iron_Man.jpg/revision/latest?cb=20110914072258","universe":"Marvel"},{"name":"Juggernaut","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/1/16/MVC2_Juggernaut.jpg/revision/latest?cb=20110914072311","universe":"Marvel"},{"name":"Magneto","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/c/c2/MVC2_Magneto.jpg/revision/latest?cb=20110914072326","universe":"Marvel"},{"name":"Marrow","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/6/65/MVC2_Marrow.jpg/revision/latest?cb=20110914072340","universe":"Marvel"},{"name":"Omega Red","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/d/dc/MVC2_Omega_Red.jpg/revision/latest?cb=20110914072357","universe":"Marvel"},{"name":"Psylocke","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/4/4b/MVC2_Psylocke.jpg/revision/latest?cb=20110914063058","universe":"Marvel"},{"name":"Rogue","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/4/49/MVC2_Rogue.jpg/revision/latest?cb=20110914063849","universe":"Marvel"},{"name":"Sabretooth","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/0/0e/MVC2_Sabretooth.jpg/revision/latest?cb=20110914064154","universe":"Marvel"},{"name":"Sentinel","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/8/87/MVC2_Sentinel.jpg/revision/latest?cb=20110914072425","universe":"Marvel"},{"name":"Shuma-Gorath","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/1/13/MVC2_Shuma-Gorath.jpg/revision/latest?cb=20110914072443","universe":"Marvel"},{"name":"Silver Samurai","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/2/23/MVC2_Silver_Samurai.jpg/revision/latest?cb=20110914064249","universe":"Marvel"},{"name":"Spider-Man","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/9/99/MVC2_Spider-Man.jpg/revision/latest?cb=20110914072459","universe":"Marvel"},{"name":"Spiral","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/3/3f/MVC2_Spiral.jpg/revision/latest?cb=20110914064425","universe":"Marvel"},{"name":"Storm","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/9/9a/MVC2_Storm.jpg/revision/latest?cb=20110914072530","universe":"Marvel"},{"name":"Thanos","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/9/92/MVC2_Thanos.jpg/revision/latest?cb=20110914064530","universe":"Marvel"},{"name":"Venom","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/a/a7/MVC2_Venom.jpg/revision/latest?cb=20110914064640","universe":"Marvel"},{"name":"War Machine","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/e/e1/MVC2_War_Machine.jpg/revision/latest?cb=20110914064752","universe":"Marvel"},{"name":"Wolverine","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/1/15/MVC2_Wolverine.jpg/revision/latest?cb=20110914072738","universe":"Marvel"},{"name":"Wolverine, Bone","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/2/25/MVC2_Wolverine_Bone.jpg/revision/latest?cb=20110914072750","universe":"Marvel"},{"name":"Akuma","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/7/77/MVC2_Gouki.jpg/revision/latest?cb=20110923045906","universe":"Capcom"},{"name":"Amingo","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/8/8b/MVC2_Amingo.jpg/revision/latest?cb=20110923050130","universe":"Capcom"},{"name":"Anakaris","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/4/4d/MVC2_Anakaris.jpg/revision/latest?cb=20110923050216","universe":"Capcom"},{"name":"Baby Bonnie Hood","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/a/a5/MVC2_B.B.Hood.jpg/revision/latest?cb=20110923050312","universe":"Capcom"},{"name":"Cammy","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/3/35/MVC2_Cammy.jpg/revision/latest?cb=20110923050348","universe":"Capcom"},{"name":"Captain Commando","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/4/4f/MVC2_Captain_Commando.jpg/revision/latest?cb=20110923045946","universe":"Capcom"},{"name":"Charlie","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/4/4b/MVC2_Nash.jpg/revision/latest?cb=20110923050615","universe":"Capcom"},{"name":"Chun-Li","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/8/87/MVC2_Chun-Li.jpg/revision/latest?cb=20110923050630","universe":"Capcom"},{"name":"Dan","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/8/8c/MVC2_Dan.jpg/revision/latest?cb=20110923050740","universe":"Capcom"},{"name":"Dhalsim","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/c/ca/MVC2_Dhalsim.jpg/revision/latest?cb=20110923050938","universe":"Capcom"},{"name":"Felicia","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/b/b2/MVC2_Felicia.jpg/revision/latest?cb=20110923050839","universe":"Capcom"},{"name":"Guile","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/a/a7/MVC2_Guile.jpg/revision/latest?cb=20110923051156","universe":"Capcom"},{"name":"Hayato","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/c/c3/MVC2_Hayato.jpg/revision/latest?cb=20110923051231","universe":"Capcom"},{"name":"Jill","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/d/d2/MVC2_Jill.jpg/revision/latest?cb=20110923051339","universe":"Capcom"},{"name":"Jin","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/a/a1/MVC2_Jin.jpg/revision/latest?cb=20110923051434","universe":"Capcom"},{"name":"Ken","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/9/9c/MVC2_Ken.jpg/revision/latest?cb=20110923051520","universe":"Capcom"},{"name":"Vega","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/f/f2/MVC2_Vega.jpg/revision/latest?cb=20110923051559","universe":"Capcom"},{"name":"Mega Man","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/8/8d/MVC2_Megaman.jpg/revision/latest?cb=20110923051647","universe":"Capcom"},{"name":"Morrigan","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/2/26/MVC2_Morrigan.jpg/revision/latest?cb=20110923051714","universe":"Capcom"},{"name":"Roll","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/f/f1/MVC2_Roll.jpg/revision/latest?cb=20110923051734","universe":"Capcom"},{"name":"Ruby Heart","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/4/4d/MVC2_Ruby_Heart.jpg/revision/latest?cb=20110923051812","universe":"Capcom"},{"name":"Ryu","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/5/5b/MVC2_Ryu.jpg/revision/latest?cb=20110923051830","universe":"Capcom"},{"name":"Sakura","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/e/ec/MVC2_Sakura.jpg/revision/latest?cb=20110923052012","universe":"Capcom"},{"name":"Servbot","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/4/49/MVC2_Servbot.jpg/revision/latest?cb=20110923052044","universe":"Capcom"},{"name":"SonSon","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/4/46/MVC2_Sonson.jpg/revision/latest?cb=20110923052114","universe":"Capcom"},{"name":"Strider","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/1/15/MVC2_Strider_Hiryu.jpg/revision/latest?cb=20110923051909","universe":"Capcom"},{"name":"Tron Bonne","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/7/7f/MVC2_T.Bonne.jpg/revision/latest?cb=20110923052135","universe":"Capcom"},{"name":"Zangief","head_shot":"https://vignette.wikia.nocookie.net/marvelvscapcom/images/2/24/MVC2_Zangief.jpg/revision/latest?cb=20110923052205","universe":"Capcom"},{"name":"Dalvin ","head_shot":"http://dalvin.net/profile.jpg","universe":"Marvel"}]

describe("[RENDER] Calculator â€“ Snapshot test", ()=> {
    test("Matches the snapshot", ()=>{
        const leaderboard = create(<CharInfo characters={testData} isLoading={false}/>)
        expect(leaderboard.toJSON()).toMatchSnapshot()
    })
})
