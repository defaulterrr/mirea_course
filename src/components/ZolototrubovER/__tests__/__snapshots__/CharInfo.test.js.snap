// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Calculator â€“ Snapshot test Matches the snapshot 1`] = `
<div
  className="Characters"
>
  <p>
    Marvel vs. Capcom 2 character info
  </p>
  <table
    className="CharTable"
  >
    <th>
      Character
    </th>
    <th>
      Icon
    </th>
    <th>
      Universe
    </th>
    <th>
      Info
    </th>
  </table>
</div>
`;
